src/hotspot/share/runtime/globals.hpp

```diff
@@ -2027,6 +2027,16 @@ const intx ObjectAlignmentInBytes = 8;
              "Use the FP register for holding the frame pointer "           \
              "and not as a general purpose register.")                      \
                                                                             \
+  product(intx, LogAsyncInterval, 300,                                      \
+          "Milliseconds between asynchronous log flushing")                 \
+                                                                            \
+  product(bool, AsyncLogging, false,                                        \
```

Comment:
I think this option is not needed - `async` should be set to `false` by default, and we should control it through `-Xlog` option like other log output options (e.g. `filecount`).

Reply:
It's possible that a Java process have multiple file-based outputs. A global option `AsyncLogging` can set them all Otherwise, developers have to set async=true individually. It's part of CSR, right?

Reply:
Yes, it's part of CSR.

I think it is prefer to set `async` to false by default because it should be treated same with `file` / `filecount` / `filesize` on the logger. The user should add `async=true` to the logger what the user want to set to.

Reply:
oh, I don't know this constraint.  I will add a constraint check for the option.

Reply:
add  a constraint for it. now it will get the error message for 13. 
```
java -XX:LogAsyncInterval=13 -Xlog:gc:file=/tmp/class.log::async=true -version
LogAsyncInterval (13) must be evenly divisible by PeriodicTask::interval_gran (10)
Improperly specified VM option 'LogAsyncInterval=13'
Error: Could not create the Java Virtual Machine.
Error: A fatal exception has occurred. Program will exit.
```